### ISSUE TITLE
GRADIENT_CLIPPING_NORM: 0 zeroes all gradients instead of disabling clipping

### ISSUE THREAD

**Issue by smoothyy3** [mspitzna/NCAtorch#4]
### Expected behavior:

Config documents "0" as the way to disable gradient clipping ([nca/utils/config.py:151](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/nca/utils/config.py#L151), [config/template_config.yaml:93](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/config/template_config.yaml#L93), and [config/emoji_config.yaml:41](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/config/emoji_config.yaml#L41)). 

### Actual behavior:
Setting to zero instead multiplies every gradient by zero, so the model never updates.
It fails quietly, no warning or exception and the returned total_norm is the pre-clip value, so gradient norm logging looks healthy. Only visible symptom is a flat loss from step 0.

### Root cause:
`clip_grad_norm_` is called unconditionally at [base_trainer.py:359-371](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/nca/training/trainers/base_trainer.py#L359-L371), and [adversarial_trainer.py:149](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/nca/training/trainers/adversarial_trainer.py#L149) and [:157](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/nca/training/trainers/adversarial_trainer.py#L157).

**In torch/nn/utils/clip_grad.py:165-182:**

```
clip_coef = max_norm / (total_norm + 1e-6) -> becomes 0
clamp(..., max=1.0) -> leaves it at 0
g.mul_(0) -> zeros the gradients
```

### Reproduction:

*Minimal, PyTorch 2.11 only*
```python
p = torch.nn.Parameter(torch.ones(3)); p.grad = torch.tensor([3., 4., 5.])
torch.nn.utils.clip_grad_norm_([p], max_norm=0)
# p.grad -> [0.0, 0.0, 0.0]
```

*In the framework*
20 steps with the backbone config gives max |delta_w| = 2.836e-02 at GRADIENT_CLIPPING_NORM=1.0 and exactly 0.0 at 0.

### Impact:
No shipped config uses 0, this only affects users who follow the docs. Noticed while porting Med-NCA (trains without clipping).

### Suggested fix:
maybe guard the call sites with `if norm > 0: `, matching the convention already used for `NOISE_INJECTION` and `FIRE_RATE` at [model_factory.py:64-67](https://github.com/mspitzna/NCAtorch/blob/bad1848e97b1962f2c1e1204e3cd0d672e249a65/nca/core/models/model_factory.py#L64-L67) (or just change the doc strings to say that a large value disables it).

**Comment by mspitzna** [comment]
Thanks for the detailed report and reproduction! 
`clip_grad_norm_` interprets `max_norm=0` literally and zeros the gradients, which conflicts with our documented convention that 0 disables clipping.

I’ve fixed this by only calling `clip_grad_norm_` when `GRADIENT_CLIPPING_NORM > 0`, covering both the standard trainer and adversarial critic paths. I also added tests confirming that:
- 0 leaves gradients unchanged.
- Positive values still clip gradients normally.


Thanks again for catching this! Especially the quiet failure mode.

Fixed in 939b54e791918216d57300f37780032a7b31506e
